popup.html

```diff
@@ -116,6 +116,7 @@ <h4 data-i18n="cookies_general"></h4>
 									<span data-text="domain" class="domainName"></span>
 									<span data-text="cookienum" class="cookiesNumber"></span>
 								</div>
+								<button role="checkbox" data-action="whitelist-cookie-domain" data-i18n="whitelistCookieDomain" data-info="true" class="icon whitelist"></button>
```

Comment:
Regarding `data-i18n`: Currently there is no easy way to set the title of the `title` attribute of the element using HTML.

`data-i18n` is being used to populate the content of the item it's set on, see -> https://github.com/Manvel/Privacy-Manager/blob/master/js/ui/common.js#L173 . But even in this case the content of `<button>` will not be populated because we load items dynamically and call i18n function on `DOMContentLoaded ` event which happens before we populate the items.

I think what you want(ed) to do here is to add the `whitelistCookieDomain` string during [table list item load](https://github.com/Manvel/Privacy-Manager/blob/master/js/ui/common.js#L533).

I would imagine that we might want to have is a `[data-i18n-title]` attribute which should handle the title population, but for dynamic items we might want to setup a cache.

We can either tackle that in current review or separately.

Reply:
yeah, i hoped it would work just like the regular (i) info buttons when you over over it

Reply:
not exactly sure how i should implement this though

Reply:
> not exactly sure how i should implement this though

## Solution 1
I can see two ways how it's possible to achieve that, there is `_itemFromTmpl` and `_updateListElem` methods, the logic can be implemented in either of those I assume.

I would imagine something like this:

```
const listElement = ...;

for (const i18nTitleElement of listElement.querySelectorAll("[data-i18n-title]"))
  i18nTitleElement.title = getMsg(i18nTitleElement.dataset.i18nTitle);
```

regarding cache: I would assume most of the browsers do tackle the caching, but alternatively that can be implemented in `getMsg` in `utils.js`, maybe smth like this(The logic can probably be simplified):

```
function getMsg(text, cache)
{
  let result = msgsCache[text];
  if (result)
    return result;
  else
    result = chrome.i18n.getMessage(text) || text;
  if (cache && !(text in msgsCache))
    msgsCache[text] = text;
  return result;
}
```

## Solution 2

Maybe we can also do something even better, by passing the actual text to the `itemObj` by introducing new type of data, in order to populate the text Table List is using `data-text` attribute to locate the element and update the text, so we will need to introduce something like this, considering https://github.com/Manvel/Privacy-Manager/blob/master/js/ui/tab_cookies.js#L374 :

```
function createDomainObj(domain, cookienum, whitelist = false)
  {
    return {
      dataset: {
        access: domain,
        whitelist: whitelist
      },
      texts: {
        domain: domain,
        cookienum: cookienum + " Cookies"
      },
     titles: {
        whitelist: whitelistCookieDomain
     }
    };
  }
```

And in the https://github.com/Manvel/Privacy-Manager/blob/master/js/ui/common.js#L699 

```
var titleObjs = itemObj.titles;
for (var title in titleObjs)
{
  var titleElement = listElem.querySelector("[data-text='"+ title +"']");
  if (titleElement)
    textElement.title = titleObjs[title];
}
```

And in this case we don't need to Cache anything and the logic should be consistent.
(The logic also can be simplified)

I like 2-nd solution more than 1-st one.

Reply:
updated code, now message displays correctly when hovering over the whitelist buttons
used the 2nd solution you proposed for this, seems to work well